src/app/components/side-nested-menu/__item/side-nested-menu__item.jsx

```diff
@@ -48,15 +49,18 @@ class SideNestedMenu__Item extends React.Component {
       nestedRoutePresets,
     } = this.props;
 
-    const selected = matchRoute(route, location.pathname) || nestedRoutePresets.includes(location.pathname);
+    const selected = hasSubmenu
+      ? matchRoute(route, location.pathname) || nestedRoutePresets.includes(location.pathname)
+      : matchRoute(route, location.pathname, true);
```

Comment:
This will definitely solve the problem for menu items at "action" level, but I could still get it broken for "resource" level.

This requires some additional explanations, I suppose.

As an example, we have the next group:

```markdown
# Group Группа с якорем

<!-- anchor: my-group -->

## Ресурс с якорем [/resource-with-anchor]

<!-- anchor: my-resource -->

### Первый запрос с якорем [GET]

<!-- anchor: my-resource-action -->

+ Response 200 (application/json)

### Второй запрос с якорем [POST]

<!-- anchor: my-resource-action-two -->

+ Response 200 (application/json)

## Ресурс с якорем 2 [/resource-with-anchor-two]

<!-- anchor: my-resource-two -->

### Первый запрос с якорем [GET]

<!-- anchor: my-resource-two-action -->

+ Response 200 (application/json)

### Второй запрос без якоря [POST]
```

Put this group in your test APIB doc and select `my-resource-two-action` (the first action of the second resource).
What is highlighted now? Second resource (correct), first action of second resource (correct), first resource (oops, wrong).

I'd tried to imagine in my head how to solve this problem and then it dawned on me. If we have such thing as [nestedRoutePresets](https://github.com/funbox/blueprinter/blob/master/src/app/common/utils/helpers/parse-source-file.js#L67), why don't we want just to have `nestedRoutes`?

So, we can just get rid of `nestedRoutePresets`, put all nested routes of a group (for example) to `group.nestedRoutes` (such array will contain routes of all resources and all actions in that group). And then we just need to check if the current route belongs to `nestedRoutes` array or it exactly match the route of current item.

Yes, this may be excessive if you think about array sizes, but:
1) we have all information at the parsing stage (just replace `group.nestedRoutePresets.push(action.routePreset);` with `group.nestedRoutes.push(action.route);`
2) logic behind the items selection will be pretty simple

But with this approach we should be super confident that during location change not all of menu items will be searching for the next location in their `nestedRoutes` array. There is already a [protection](https://github.com/funbox/blueprinter/blob/master/src/app/components/side-nested-menu/__item/side-nested-menu__item.jsx#L25) against unwanted render and I hope it will continue to work.


All of this stuff is just my vision and you are not required to follow it =)

Reply:
@Ge11ert, @bananafishh, it would be great if you decide together whether or not the `nestedRoutePresets` behavior should be changed in this PR.

I don't have any strong opinion here, however I believe that the less code we have the better. So, if we can get rid of something, it would be great to do. The question is — should it be done in this PR, or will someone do it later?

Reply:
The idea of my comment above is that current solution fixes only one piece of a problem while there is another piece (see my example doc).

If @bananafishh can fix doubled highlighting both for actions and resources without involving `nestedRoutes` thing, so be it. But in case any problem with it, we could try with my offer.

> should it be done in this PR, or will someone do it later

Depends on the final solution. May be it is the only way to correctly handle all cases with items selection, then it should be done here. If the final solution is somehow different, we could do it later.

Reply:
@Ge11ert , thanks a lot for your advice, but it turned out to be too hard for me to understand, and I couldn't use it properly) However, I think I managed to improve my code following my previous line of thinking. Please, check it one more time)